src/OrchardCore.Modules/OrchardCore.Forms/Views/Items/InputPart.cshtml

```diff
@@ -14,7 +14,7 @@
     {
         if (Model.Value.Type == "checkbox")
         {
-            isChecked = fieldEntry.AttemptedValue == fieldValue;
+            isChecked = fieldEntry.AttemptedValue == (fieldValue ?? "on");
```

Comment:
"on" is the value for a checkbox that doesn't specify a value. [MDN checkbox docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox)

If you don't specify "default value" on the InputPart what ends up being submitted for the checkbox is "on".

Reply:
If this was using the .NET input tag helper it would generate two `<inputs>`, one with a value of "true" and another hidden one with a value of "false".

Reply:
My point was that it was not clear. The comment that was added is all I needed to see :)

Reply:
My point was that it was not clear. The comment that was added is all I needed to see :)